ibis/expr/types/numeric.py

```diff
@@ -112,7 +112,57 @@ def clip(
                 "at least one of lower and upper must be provided"
             )
 
-        return ops.Clip(self, lower, upper).to_expr()
+        if lower is None:
+            op = ops.ClipUpper(self, upper)
+        elif upper is None:
+            op = ops.ClipLower(self, lower)
+        else:
+            op = ops.Clip(self, lower, upper)
+        return op.to_expr()
+
+    def clip_lower(
```

Comment:
Is this intentionally a public API? `clip_lower`'s functionality can be captured by inspecting the upper and lower bounds and producing the expression you want to produce.

You can override ibis-substrait compilation of `ops.Clip` and define your own translation rules for `ClipLower` and `ClipUpper` and then this doesn't need to live upstream.

Reply:
Not related to UDT; please skip.